src/lib/remark-plugins/rehype-sanitize/index.ts

```diff
@@ -19,20 +19,38 @@ const schema = deepmerge(defaultSchema, {
 	 * as it's typically used on plain markdown or HTML, not MDX, I think.
 	 */
 	tagNames: [...defaultSchema.tagNames, 'inlineCode'],
-	/**
-	 * We enable class names for all elements.
-	 *
-	 * This allows "paragraph custom alerts" as well as our `rehype-prism`
-	 * syntax highlighting to function as expected.
-	 * - Our paragraph custom alerts: https://github.com/hashicorp/remark-plugins/tree/main/plugins/paragraph-custom-alerts
-	 * - Our `rehype-prism` setup: https://github.com/hashicorp/web-platform-packages/tree/main/packages/code-highlighting
-	 *
-	 * Note: we could create more strictly scope rules here, instead of allowing
-	 * all class name values on all elements. However, there doesn't seem to be
-	 * any security concern here, and a stricter set of specific strings and
-	 * regex patterns for classNames would be much more difficult to maintain.
-	 */
-	attributes: { '*': ['className'] },
+	attributes: {
+		'*': [
+			/**
+			 * We enable class names for all elements.
+			 *
+			 * This allows "paragraph custom alerts" as well as our `rehype-prism`
+			 * syntax highlighting to function as expected.
+			 * - Our paragraph custom alerts: https://github.com/hashicorp/remark-plugins/tree/main/plugins/paragraph-custom-alerts
+			 * - Our `rehype-prism` setup: https://github.com/hashicorp/web-platform-packages/tree/main/packages/code-highlighting
+			 *
+			 * Note: we could create more strictly scope rules here, instead of allowing
+			 * all class name values on all elements. However, there doesn't seem to be
+			 * any security concern here, and a stricter set of specific strings and
+			 * regex patterns for classNames would be much more difficult to maintain.
+			 */
+			'className',
+			/**
+			 * `data-text-content` is used for our `remark-plugin-anchor-links-data`.
+			 * We enable data-text-content for all elements (enabling it for heading
+			 * elements only would be less clean, and no more safe).
+			 *
+			 * We use config here to ensure that only alphanumeric characters are
+			 * present. Any `data-text-content` attribute with any other characters
+			 * will be removed completely.
+			 *
+			 * In the future, we could safely expand this to 'data*' to avoid having
+			 * to maintain and reason about specific use cases.
+			 * Ref: https://github.com/syntax-tree/hast-util-sanitize#attributes
+			 */
+			['data-text-content', /[\w\-\s]+/],
```

Comment:
Tried to be thorough here with comments.

In short the upgraded `rehype-sanitize` dependency lets us only allow `data-text-content` properties if they have strictly alphanumeric values. For example, if our plugin code for any reason failed to remove some potentially dangerous characters, or if a user decided to try to add them in, `rehype-sanitize` is still the last thing we run, so would filter those entire properties out.

Due to the test complexity issues I encountered (I think related to `rehype-sanitize` being ESM-only), might need to downgrade again if we can't find a good solution for tests, and find a different solution here.